services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -367,4 +456,108 @@ private void addValidIndividualRequest(final IBulkSubmission bulkSubmission, fin
         bulkSubmission.addIndividualRequest(individualRequest);
     }
 
+
+    @Test
+    public void setErrorMessagesCacheTest() throws Exception {
+        IGenericDao genericDaoMock = mock(GenericDao.class);
+        IBulkCustomerDao bulkCustomerDaoMock = mock(BulkCustomerDao.class);
+        ITargetApplicationDao targetApplicationDaoMock = mock(TargetApplicationDao.class);
+        IndividualRequestsXmlParser individualRequestsXmlParserMock = mock(IndividualRequestsXmlParser.class);
+        IMessagingUtility messagingUtility = mock(MessagingUtility.class);
+        ISdtBulkReferenceGenerator sdtBulkReferenceGeneratorMock = mock(SdtBulkReferenceGenerator.class);
+        ICacheable errorMessagesCache = Mockito.mock(ICacheable.class);
+
+
+        BulkSubmissionService bulkSubmission = new BulkSubmissionService(
+            genericDaoMock,bulkCustomerDaoMock, targetApplicationDaoMock,individualRequestsXmlParserMock,
+            messagingUtility,sdtBulkReferenceGeneratorMock,errorMessagesCache);
+
+
+        bulkSubmission.setErrorMessagesCache(errorMessagesCache);
+
+        Field errorMessagesCacheField = BulkSubmissionService.class.getDeclaredField("errorMessagesCache");
+        errorMessagesCacheField.setAccessible(true);
+        ICacheable result = (ICacheable) errorMessagesCacheField.get(bulkSubmission);
+
+        assertEquals(errorMessagesCache, result, "The errorMessagesCache should be set correctly");
+    }
+
+
+    @Test
+    public void setMockSdtBulkReferenceGeneratorTest() throws Exception {
+
+        bulkSubmissionService.setSdtBulkReferenceGenerator(mockSdtBulkReferenceGenerator);
+
+        Object result = this.getAccessibleField(BulkSubmissionService.class,"sdtBulkReferenceGenerator",
+                                                SdtBulkReferenceGenerator.class,bulkSubmissionService);
+
+        assertEquals(mockSdtBulkReferenceGenerator, result, "The sdtBulkReferenceGenerator should be set correctly");
```

Comment:
Perhaps change message to "The sdtBulkReferenceGenerator was not set to the expected object."

Reply:
done